File: api/internal/entity/CourseMember.go
package entity

import (
	"gorm.io/gorm"
	"time"
)

type CourseMember struct {
	gorm.Model
	CourseId   uint
	Course     *Course
	StudentId  uint
	Student    *User
	EnrolledOn *time.Time
}

func NewCourseMember(courseId uint, studentId uint) *CourseMember {
	now := time.Now()
	return &CourseMember{
		CourseId:   courseId,
		StudentId:  studentId,
		EnrolledOn: &now,
	}
}

File: api-v2/smart-elearning/src/main/java/com/smartelearning/smartelearning/entity/CourseMember.java
/*
 *  CourseMember
 *  @author: minhhieuano
 *  @created 7/10/2025 3:58 PM
 * */


package com.smartelearning.smartelearning.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.Instant;

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class CourseMember extends BaseEntity {
    private String courseId;
    private String studentId;
    private Instant enrolledOn;
}
